src/hooks/useActiveSubscription.tsx

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useContext } from 'react';
+import getActiveSubscription from 'actions/getActiveSubscription';
+import useAsync from 'hooks/useAsync';
+import { TokenContext } from 'hooks/useCustomerToken';
+import { components } from '../apiTypes';
+
+type PricePlan = components['schemas']['PricePlan'];
+
+export type UseActiveSubscriptionReturnType = {
+  result: PricePlan | null;
+  loading: boolean;
+  error: unknown;
+};
```

Comment:
This type is actually less strict than `UseAsyncReturnType`. This type would allow for something like this: 
```ts
{ loading: false, result: null, error: null }
```
which wouldn't be allowed under `UseAsyncReturnType`. 

I think you might want to export `UseAsyncReturnType` and use it here:
```ts
export type UseActiveSubscriptionReturnType = UseAsyncReturnType<PricePlan, unknown>
```

It might be worth it to have `Error` default to `unknown` in `UseAsyncReturnType`, too:

```ts
type UseAsyncReturnType<Result, Error = unknown> = ...
```

Reply:
yeah, this makes sense, thank you!